v4/export/sql.go

```diff
@@ -277,3 +313,112 @@ func simpleQuery(db *sql.DB, sql string, handleOneRow func(*sql.Rows) error) err
 	}
 	return nil
 }
+
+func pickupPossibleField(dbName, tableName string, db *sql.DB, conf *Config) (string, error) {
+	// If detected server is TiDB, try using _tidb_rowid
+	if conf.ServerInfo.ServerType == ServerTypeTiDB {
+		ok, err := SelectTiDBRowID(db, dbName, tableName)
+		if err != nil {
+			return "", nil
+		}
+		if ok {
+			return "_tidb_rowid", nil
+		}
+	}
+	// try to use pk
+	fieldName, err := GetPrimaryKeyName(db, dbName, tableName)
+	if err != nil {
+		return "", err
+	}
+	// try to use first uniqueIndex
+	if fieldName == "" {
+		fieldName, err = GetUniqueIndexName(db, dbName, tableName)
+	}
+
+	// there is no proper index
+	if fieldName == "" {
+		return "", nil
+	}
+
+	query := fmt.Sprintf("SELECT DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS "+
+		"WHERE TABLE_NAME = `%s` AND COLUMN_NAME = `%s`", tableName, fieldName)
+	var fieldType string
+	row := db.QueryRow(query)
+	err = row.Scan(&fieldType)
```

Comment:
ditto, be care of `sql.ErrNoRows`. https://blog.csdn.net/rocky0503/article/details/80884678

Reply:
Fixed